### ISSUE TITLE
 sealos  run   labring/apisix:2.15.0    --cmd "helm install....." 后cmd的命令没有覆盖原有镜像里的cmd，而且apisix2.15.0的镜像也部署不成功

### ISSUE THREAD

**Issue by ouweijun1** [labring/sealos#1969]
[root@master01 ~]# sealos  run   labring/apisix:2.15.0    --cmd "helm install  apisix charts/apisix  --set dashboard.enabled=true,ingress-controller.enabled=true, -n ingress-apisix --create-namespace"

2022-10-27T16:34:40 info guest cmd is helm upgrade -i apisix charts/apisix --set gateway.type=NodePort,ingress-controller.enabled=true,ingress-controller.config.apisix.serviceNamespace=ingress-apisix -n ingress-apisix --create-namespace
Release "apisix" does not exist. Installing it now.
Error: template: apisix/templates/serviceaccount.yaml:17:14: executing "apisix/templates/serviceaccount.yaml" at <.Values.serviceAccount.create>: nil pointer evaluating interface {}.create
2022-10-27T16:34:52 error Applied to cluster error: exit status 1

- Kubernetes version:  v1.22.0
- Sealos version:  4.1.3
- Operating system: eg. centos7

**Comment by fanux** [comment]
1. check .sealos/default/Clusterfile
2. cd [the apisix rootfs] , like /var/lib/containers/../merged
3. check the helm and the CMD command.

**Comment by fanux** [comment]
```
helm upgrade -i apisix charts/apisix --set gateway.type=NodePort,ingress-controller.enabled=true,ingress-controller.config.apisix.serviceNamespace=ingress-apisix -n ingress-apisix --create-namespace","helm upgrade -i apisix-dashboard charts/apisix-dashboard -n ingress-apisix --create-namespace --set service.type=NodePort
```
can you exec the command manully?

**Comment by ouweijun1** [comment]
> ```
> helm upgrade -i apisix charts/apisix --set gateway.type=NodePort,ingress-controller.enabled=true,ingress-controller.config.apisix.serviceNamespace=ingress-apisix -n ingress-apisix --create-namespace","helm upgrade -i apisix-dashboard charts/apisix-dashboard -n ingress-apisix --create-namespace --set service.type=NodePort
> ```
> 
> can you exec the command manully?

 主要是我run的时候在后面加的cmd好像没有作用，没有覆盖掉原有的

**Comment by ouweijun1** [comment]
> 

这个执行成功了

**Comment by cuisongliu** [comment]
> > ```
> > helm upgrade -i apisix charts/apisix --set gateway.type=NodePort,ingress-controller.enabled=true,ingress-controller.config.apisix.serviceNamespace=ingress-apisix -n ingress-apisix --create-namespace","helm upgrade -i apisix-dashboard charts/apisix-dashboard -n ingress-apisix --create-namespace --set service.type=NodePort
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > can you exec the command manully?
> 
> 
> 主要是我run的时候在后面加的cmd好像没有作用，没有覆盖掉原有的

try using sealos 4.1.4-rc1 ?

**Comment by ouweijun1** [comment]
> > > ```
> > > helm upgrade -i apisix charts/apisix --set gateway.type=NodePort,ingress-controller.enabled=true,ingress-controller.config.apisix.serviceNamespace=ingress-apisix -n ingress-apisix --create-namespace","helm upgrade -i apisix-dashboard charts/apisix-dashboard -n ingress-apisix --create-namespace --set service.type=NodePort
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > can you exec the command manully?
> > 
> > 
> > 主要是我run的时候在后面加的cmd好像没有作用，没有覆盖掉原有的
> 
> try using sealos 4.1.4-rc1 ?
这个
[root@master01 ~]# sealos  version
{"gitVersion":"4.1.3","gitCommit":"b2ba9705","buildDate":"2022-09-06T06:04:14Z","goVersion":"go1.19","compiler":"gc","platform":"linux/amd64"}

**Comment by willzhang** [comment]
same with 4.1.3
```
root@node1:~# sealos run labring/apisix:2.15.0 --cmd "ls"
2022-10-28T12:46:05 info start to install app in this cluster
Resolving "labring/apisix" using unqualified-search registries (/etc/containers/registries.conf)
Trying to pull docker.io/labring/apisix:2.15.0...
Getting image source signatures
Copying blob f88e678d3819 done  
Copying config f37576bd51 done  
Writing manifest to image destination
Storing signatures
f37576bd51de270cc81e312a3c969e19716ffe3e95006250f8f56279903b913c
default-zxto356c
2022-10-28T12:46:31 info guest cmd is helm upgrade -i apisix charts/apisix --set gateway.type=NodePort,ingress-controller.enabled=true,ingress-controller.config.apisix.serviceNamespace=ingress-apisix -n ingress-apisix --create-namespace
Release "apisix" does not exist. Installing it now.
NAME: apisix
LAST DEPLOYED: Fri Oct 28 12:46:37 2022
NAMESPACE: ingress-apisix
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
1. Get the application URL by running these commands:
  export NODE_PORT=$(kubectl get --namespace ingress-apisix -o jsonpath="{.spec.ports[0].nodePort}" services apisix-gateway)
  export NODE_IP=$(kubectl get nodes --namespace ingress-apisix -o jsonpath="{.items[0].status.addresses[0].address}")
  echo http://$NODE_IP:$NODE_PORT
2022-10-28T12:46:45 info guest cmd is helm upgrade -i apisix-dashboard charts/apisix-dashboard -n ingress-apisix --create-namespace --set service.type=NodePort
Release "apisix-dashboard" does not exist. Installing it now.
NAME: apisix-dashboard
LAST DEPLOYED: Fri Oct 28 12:46:45 2022
NAMESPACE: ingress-apisix
STATUS: deployed
REVISION: 1
NOTES:
1. Get the application URL by running these commands:
  export NODE_PORT=$(kubectl get --namespace ingress-apisix -o jsonpath="{.spec.ports[0].nodePort}" services apisix-dashboard)
  export NODE_IP=$(kubectl get nodes --namespace ingress-apisix -o jsonpath="{.items[0].status.addresses[0].address}")
  echo http://$NODE_IP:$NODE_PORT
2022-10-28T12:46:46 info succeeded install app in this cluster
2022-10-28T12:46:46 info start to scale this cluster
2022-10-28T12:46:46 info succeeded in scaling this cluster: no change nodes
2022-10-28T12:46:46 info 
      ___           ___           ___           ___       ___           ___
     /\  \         /\  \         /\  \         /\__\     /\  \         /\  \
    /::\  \       /::\  \       /::\  \       /:/  /    /::\  \       /::\  \
   /:/\ \  \     /:/\:\  \     /:/\:\  \     /:/  /    /:/\:\  \     /:/\ \  \
  _\:\~\ \  \   /::\~\:\  \   /::\~\:\  \   /:/  /    /:/  \:\  \   _\:\~\ \  \
 /\ \:\ \ \__\ /:/\:\ \:\__\ /:/\:\ \:\__\ /:/__/    /:/__/ \:\__\ /\ \:\ \ \__\
 \:\ \:\ \/__/ \:\~\:\ \/__/ \/__\:\/:/  / \:\  \    \:\  \ /:/  / \:\ \:\ \/__/
  \:\ \:\__\    \:\ \:\__\        \::/  /   \:\  \    \:\  /:/  /   \:\ \:\__\
   \:\/:/  /     \:\ \/__/        /:/  /     \:\  \    \:\/:/  /     \:\/:/  /
    \::/  /       \:\__\         /:/  /       \:\__\    \::/  /       \::/  /
     \/__/         \/__/         \/__/         \/__/     \/__/         \/__/

                  Website :https://www.sealos.io/
                  Address :github.com/labring/sealos
```

but the image have no problem
```
root@node1:~# kubectl -n ingress-apisix  get pods 
NAME                                         READY   STATUS    RESTARTS       AGE
apisix-6fcf4bc5fc-gwzpf                      1/1     Running   0              2m47s
apisix-dashboard-764fb6df8d-t2clc            1/1     Running   4 (101s ago)   2m46s
apisix-etcd-0                                1/1     Running   0              2m47s
apisix-etcd-1                                1/1     Running   0              2m47s
apisix-etcd-2                                1/1     Running   0              2m47s
apisix-ingress-controller-6cf796c6b8-8lx6p   1/1     Running   0              2m47s
root@node1:~# 
root@node1:~# 
root@node1:~# kubectl -n ingress-apisix  get pods -o yaml |grep image:
    - image: apache/apisix:2.15.0-alpine
      image: busybox:1.28
      image: sealos.hub:5000/apache/apisix:2.15.0-alpine
      image: sealos.hub:5000/library/busybox:1.28
    - image: apache/apisix-dashboard:2.13-alpine
      image: sealos.hub:5000/apache/apisix-dashboard:2.13-alpine
      image: docker.io/bitnami/etcd:3.5.4-debian-11-r14
      image: sealos.hub:5000/bitnami/etcd:3.5.4-debian-11-r14
      image: docker.io/bitnami/etcd:3.5.4-debian-11-r14
      image: sealos.hub:5000/bitnami/etcd:3.5.4-debian-11-r14
      image: docker.io/bitnami/etcd:3.5.4-debian-11-r14
      image: sealos.hub:5000/bitnami/etcd:3.5.4-debian-11-r14
      image: apache/apisix-ingress-controller:1.4.1
      image: busybox:1.28
      image: sealos.hub:5000/apache/apisix-ingress-controller:1.4.1
      image: sealos.hub:5000/library/busybox:1.28
```

**Comment by willzhang** [comment]
yes, it's fixed by 4.1.4-rc1

```
root@node1:~# sealos version
{"gitVersion":"4.1.4-rc1","gitCommit":"990b821d","buildDate":"2022-10-26T14:16:54Z","goVersion":"go1.19.2","compiler":"gc","platform":"linux/amd64"}
root@node1:~# 
root@node1:~# sealos run -f labring/apisix:2.15.0 --cmd "ls"
2022-10-28T12:52:52 info start to install app in this cluster
are you sure to override these app? Yes [y/yes], No [n/no] : y
2022-10-28T12:52:54 info pulling images [labring/apisix:2.15.0] for platform linux/amd64
f37576bd51de270cc81e312a3c969e19716ffe3e95006250f8f56279903b913c
default-zxto356c
2022-10-28T12:53:00 info guest cmd is ls
bin  charts  cri  etc  images  init.sh  Kubefile  manifests  opt  README.md  registry  scripts  statics
2022-10-28T12:53:00 info succeeded install app in this cluster
2022-10-28T12:53:00 info no nodes that need to be scaled
2022-10-28T12:53:01 info 
      ___           ___           ___           ___       ___           ___
     /\  \         /\  \         /\  \         /\__\     /\  \         /\  \
    /::\  \       /::\  \       /::\  \       /:/  /    /::\  \       /::\  \
   /:/\ \  \     /:/\:\  \     /:/\:\  \     /:/  /    /:/\:\  \     /:/\ \  \
  _\:\~\ \  \   /::\~\:\  \   /::\~\:\  \   /:/  /    /:/  \:\  \   _\:\~\ \  \
 /\ \:\ \ \__\ /:/\:\ \:\__\ /:/\:\ \:\__\ /:/__/    /:/__/ \:\__\ /\ \:\ \ \__\
 \:\ \:\ \/__/ \:\~\:\ \/__/ \/__\:\/:/  / \:\  \    \:\  \ /:/  / \:\ \:\ \/__/
  \:\ \:\__\    \:\ \:\__\        \::/  /   \:\  \    \:\  /:/  /   \:\ \:\__\
   \:\/:/  /     \:\ \/__/        /:/  /     \:\  \    \:\/:/  /     \:\/:/  /
    \::/  /       \:\__\         /:/  /       \:\__\    \::/  /       \::/  /
     \/__/         \/__/         \/__/         \/__/     \/__/         \/__/

                  Website :https://www.sealos.io/
                  Address :github.com/labring/sealos
                BuildVersion: 4.1.4-rc1-990b821d
```
